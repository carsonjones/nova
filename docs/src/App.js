import React, {Component} from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';

class App extends Component {
  render() {
    return (
      <div style={{
        fontFamily: '"Quicksand", sans-serif',
        fontSize: '1.3rem',
        background: '#717C89',
        color: '#E8EBE4',
        display: 'flex',
        flexDirection: 'column',
        minHeight: '100vh',
      }}>
        <Header />
        <Main />
        <Footer />
      </div>
    );
  }
}

export default App;
