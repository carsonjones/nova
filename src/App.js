import React from 'react'
import {
  colors,
  fontFamilies,
  fontSizes,
  fontWeights,
  lineHeights,
} from './styleGuide'
import Header from './Header'
import Main from './Main'
import Footer from './Footer'

const App = () => ( 
  <div style={{
    background: colors.normal.black,
    color: colors.normal.white,
    fontFamily: fontFamilies.primary,
    fontSize: fontSizes.medium,
    fontWeight: fontWeights.normal,
    lineHeight: lineHeights.medium,
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
  }}>
    <Header />
    <Main />
    <Footer />
  </div>
)

export default App
