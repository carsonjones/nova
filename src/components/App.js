import React from 'react'
import {
  screenSizes,
  colors,
  fontFamilies,
  fontSizes,
  fontWeights,
  lineHeights,
  animationSpeeds,
} from '../data/styleGuide'
import Header from './Header'
import Main from './Main'

class App extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      isIntroEnded: false,
    }
    this.handleIntroEnd = this.handleIntroEnd.bind(this)
  }

  handleIntroEnd() {
    this.setState({
      isIntroEnded: true,
    })
  }

  render() {
    return (
      <div style={{
        background: colors.normal.black,
        color: colors.bright.black,
        fontFamily: fontFamilies.primary,
        fontSize: fontSizes.medium,
        fontWeight: fontWeights.light,
        lineHeight: lineHeights.medium,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        minHeight: '100vh',
      }}>
        <div style={{
          maxWidth: screenSizes.large,
        }}>
          <Header onIntroEnd={this.handleIntroEnd} />
          <div style={{
            transition: `opacity ${animationSpeeds.medium}s`,
            opacity: this.state.isIntroEnded ? 1 : 0,
          }}>
            <Main>
              {this.props.children}
            </Main>
          </div>
        </div>
      </div>
    )
  }
}

export default App
