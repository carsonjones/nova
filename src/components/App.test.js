import React from 'react'
import App from './App'
import renderer from 'react-test-renderer'

test('App renders as expected', () => {
  const tree = renderer.create(<App />)
  expect(tree).toMatchSnapshot()
})
