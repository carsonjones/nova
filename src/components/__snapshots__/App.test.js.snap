exports[`test App renders as expected 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#445660",
      "color": "#899BA6",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "\"Quicksand\", sans-serif",
      "fontSize": 18,
      "fontWeight": 300,
      "height": "100vh",
      "lineHeight": 1.5,
      "minHeight": "100vh",
      "overflowY": "hidden"
    }
  }>
  <div
    style={
      Object {
        "maxWidth": 740
      }
    }>
    <div
      style={
        Object {
          "marginTop": "30vh",
          "transition": "margin 1s ease",
          "willChange": "margin"
        }
      }>
      <div
        style={
          Object {
            "paddingLeft": 24,
            "paddingRight": 24
          }
        }>
        <header>
          <div>
            <div
              style={
                Object {
                  "color": "#7FC1CA",
                  "fontSize": 103,
                  "lineHeight": 1.25,
                  "textAlign": "center"
                }
              }>
              <div
                className="Typist ">
                <span
                  className="Cursor Cursor--blinking">
                  |
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "color": "#C5D4DD",
                  "fontSize": 22,
                  "marginBottom": 16,
                  "textAlign": "center"
                }
              } />
          </div>
        </header>
      </div>
    </div>
    <div
      style={
        Object {
          "opacity": 0,
          "transition": "opacity 1.5s ease",
          "willChange": "opacity"
        }
      }>
      <main
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "flexDirection": "column"
          }
        } />
    </div>
  </div>
</div>
`;
